src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -184,7 +184,15 @@ Handle JavaArgumentUnboxer::next_arg(BasicType expectedType) {
   JVMCI_VM_ENTRY_MARK;                                     \
   ResourceMark rm;                                         \
   bool __is_hotspot = env == thread->jni_environment();    \
-  CompilerThreadCanCallJava ccj(thread, __is_hotspot);     \
+  bool __block_can_call_java;                              \
+  if (__is_hotspot) {                                      \
+    __block_can_call_java = true;                          \
+  } else if (thread->is_Compiler_thread()) {               \
+    __block_can_call_java = CompilerThread::cast(thread)->can_call_java();  \
+  } else {                                                 \
+    __block_can_call_java = false;                         \
```

Comment:
For non-CompilerThreads, `__block_can_call_java` should be true I think since they are affected by `-Xcomp` or `-Xbatch`. A TruffleCompileThread is such a thread.

Reply:
For non-compiler thread the new value is never used because [CompilerThreadCanCallJava::update](https://github.com/openjdk/jdk/blob/f687c82ef9ede1d9d02ca0965c896bcf658c450a/src/hotspot/share/jvmci/jvmci.cpp#L58) does not modify the `CompilerThread::_can_call_java` value in this case.
However, using `true` may improve readability. I will change it.